src/irule.cpp

```diff
@@ -322,11 +325,25 @@ main( int argc, char **argv ) {
                 if ( rulegen ) {
                     if ( startsWith( buf, "INPUT" ) || startsWith( buf, "input" ) ) {
                         gotRule = 1;
-                        trimSpaces( trimPrefix( buf ) );
+                        constexpr auto skip_len{5};
+                        const auto copy_length{std::strlen(buf) + 1 - skip_len};
```

Comment:
Looks like this can be `constexpr` since `buf` is allocated on the stack.

Reply:
I think since the string in `buf` is modified in-place, it cannot be `constexpr`.

Reply:
Why does modification matter here? `buf` lives on the stack. The modification of the contents is independent of the storage.
```cpp
constexpr auto copy_length = sizeof(buf) - skip_len;
```

Reply:
I think I get you.

The original idea was to shift only the usable data in `buf`. It's why `copy_length` is not `sizeof(buf)`, and is instead `std::strlen(buf)`.

We can shift everything in `buf` to make it `constexpr`, which you're suggesting, correct?

Reply:
Hmm, let's keep what you have. I forgot that buffer is likely holding a string passed via stdin or a file.

Reply:
Marking as resolved.